accio-base/src/main/java/io/accio/base/config/ConfigManager.java

```diff
@@ -0,0 +1,345 @@
+                postgresConfig = Optional.of(result);
+                return result;
+            });
+        }
+        if (config == DuckDBConfig.class) {
+            return (T) duckDBConfig.orElseGet(() -> {
+                DuckDBConfig result = new DuckDBConfig();
+                result.setMemoryLimit(DataSize.valueOf(configs.get(DUCKDB_MEMORY_LIMIT)));
+                result.setHomeDirectory(configs.get(DUCKDB_HOME_DIRECTORY));
+                result.setTempDirectory(configs.get(DUCKDB_TEMP_DIRECTORY));
+                result.setMaxConcurrentTasks(Integer.parseInt(configs.get(DUCKDB_MAX_CONCURRENT_TASKS)));
+                result.setMaxConcurrentMetadataQueries(Integer.parseInt(configs.get(DUCKDB_MAX_CONCURRENT_QUERIES)));
+                result.setMaxCacheQueryTimeout(Integer.parseInt(configs.get(DUCKDB_MAX_CACHE_QUERY_TIMEOUT)));
+                result.setCacheTaskRetryDelay(Integer.parseInt(configs.get(DUCKDB_CACHE_TASK_RETRY_DELAY)));
+                duckDBConfig = Optional.of(result);
+                return result;
+            });
+        }
+        if (config == PostgresWireProtocolConfig.class) {
+            return (T) postgresWireProtocolConfig.orElseGet(() -> {
+                PostgresWireProtocolConfig result = new PostgresWireProtocolConfig();
+                result.setPort(configs.get(PG_WIRE_PROTOCOL_PORT));
+                result.setSslEnable(Boolean.parseBoolean(configs.get(PG_WIRE_PROTOCOL_SSL_ENABLED)));
+                result.setNettyThreadCount(Integer.parseInt(configs.get(PG_WIRE_PROTOCOL_NETTY_THREAD_COUNT)));
+                result.setAuthFile(new File(configs.get(PG_WIRE_PROTOCOL_AUTH_FILE)));
+                postgresWireProtocolConfig = Optional.of(result);
+                return result;
+            });
+        }
+        if (config == CacheStorageConfig.class &&
+                accioConfig.map(AccioConfig::getDataSourceType).stream().anyMatch(type -> type == AccioConfig.DataSourceType.BIGQUERY)) {
+            return (T) duckdbS3StyleStorageConfig.orElseGet(() -> {
+                DuckdbS3StyleStorageConfig result = new DuckdbS3StyleStorageConfig();
+                result.setEndpoint(configs.get(DUCKDB_STORAGE_ENDPOINT));
+                result.setAccessKey(configs.get(DUCKDB_STORAGE_ACCESS_KEY));
+                result.setSecretKey(configs.get(DUCKDB_STORAGE_SECRET_KEY));
+                result.setRegion(configs.get(DUCKDB_STORAGE_REGION));
+                result.setUrlStyle(configs.get(DUCKDB_STORAGE_URL_STYLE));
+                duckdbS3StyleStorageConfig = Optional.of(result);
+                return result;
+            });
+        }
+        throw new RuntimeException("Unknown config class: " + config.getName());
+    }
+
+    public synchronized void setConfigs(List<ConfigEntry> configEntries)
+    {
+        configEntries.forEach(entry -> setConfigInternal(entry.getName(), entry.getValue()));
+        resetCache();
+    }
```

Comment:
Should we update the config file?

Reply:
no, we only handle all configs in the memory layer.

Reply:
The runtime config is different with config file is weired.

Reply:
As the discussion, we will sync memory to the config file and archived the original one.